### ISSUE TITLE
Bug: QC won't start after #569

### ISSUE THREAD

**Issue by glocalglocal** [ccbogel/QualCoder#570]
Following #569  I downloaded from master. Qualcoder didn't start and I got this: 

```
PS C:\Other programs\QualCoder-master> py -m qualcoder
Traceback (most recent call last):
  File "C:\Users\user\AppData\Local\Programs\Python\Python39\lib\runpy.py", line 197, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "C:\Users\user\AppData\Local\Programs\Python\Python39\lib\runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "C:\Other programs\QualCoder-master\qualcoder\__main__.py", line 51, in <module>
    from qualcoder.cases import DialogCases
  File "C:\Other programs\QualCoder-master\qualcoder\cases.py", line 49, in <module>
    from .case_file_manager import DialogCaseFileManager
  File "C:\Other programs\QualCoder-master\qualcoder\case_file_manager.py", line 42, in <module>
    from .view_av import DialogViewAV
  File "C:\Other programs\QualCoder-master\qualcoder\view_av.py", line 96, in <module>
    from .speech_to_text import SpeechToText
  File "C:\Other programs\QualCoder-master\qualcoder\speech_to_text.py", line 33, in <module>
    import pydub
ModuleNotFoundError: No module named 'pydub'
```

I installed pydub and then got this: 
```
PS C:\Other programs\QualCoder-master> py -m qualcoder
C:\Users\user\AppData\Local\Programs\Python\Python39\lib\site-packages\pydub\utils.py:170: RuntimeWarning: Couldn't find ffmpeg or avconv - defaulting to ffmpeg, but may not work
  warn("Couldn't find ffmpeg or avconv - defaulting to ffmpeg, but may not work", RuntimeWarning)
Traceback (most recent call last):
  File "C:\Users\user\AppData\Local\Programs\Python\Python39\lib\runpy.py", line 197, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "C:\Users\user\AppData\Local\Programs\Python\Python39\lib\runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "C:\Other programs\QualCoder-master\qualcoder\__main__.py", line 51, in <module>
    from qualcoder.cases import DialogCases
  File "C:\Other programs\QualCoder-master\qualcoder\cases.py", line 49, in <module>
    from .case_file_manager import DialogCaseFileManager
  File "C:\Other programs\QualCoder-master\qualcoder\case_file_manager.py", line 42, in <module>
    from .view_av import DialogViewAV
  File "C:\Other programs\QualCoder-master\qualcoder\view_av.py", line 96, in <module>
    from .speech_to_text import SpeechToText
  File "C:\Other programs\QualCoder-master\qualcoder\speech_to_text.py", line 37, in <module>
    import speech_recognition
ModuleNotFoundError: No module named 'speech_recognition'
```

**Comment by ccbogel** [comment]
Thanks for the feedback.  I have updated the installation instructions, including the setup.py files.
manually you can do
 pip install pydub
pip install SpeechRecognition

Once installs QC will run.

The spechrecoginition will only work if you also have ffmpeg or avconv installed. You have to look up separately how to install those. If you are on Linux (Ubuntu) one of them is likely already installed.

**Comment by glocalglocal** [comment]
After installing SpeechRecognition it opened OK. Not sure where speech recognition is supposed to work. I suppose it is meant to help with transcription. So, I tried to open a linked mp3 and View returns "name 'tr-id' is not defined". Double-clicking the file returns: "UNIQUE constraint failed: source.name"

**Comment by ccbogel** [comment]
A soft launch for speech recognition, as might be a few additional things to resolve. Yes view an A/V file and providing there is not text in the transcript pressing the convert speech to text cogs button.

So, I tried to open a linked mp3 and View returns "name 'tr-id' is not defined". Double-clicking the file returns: "UNIQUE constraint failed: source.name"  - I need more information here, as I have opened a linked wav file in a new project, and this did not occur. Were there line number pointing to something? Do you have two files with the same name in the Manage files list?

**Comment by glocalglocal** [comment]
I linked a new .mp3 and opening it worked. The moment I double-click the 
pre-existing  mp3 files I get the same errors as above. It may be that 
these mp3 files were already transcribed and the associated txt files 
are not blank.

The tooltip mentions an online service that does the transcribing. Which 
one is that? I think there should be more transparency and possibly 
warnings because users may object to sharing the recordings, and at the 
very least they should agree to T&Cs.

**Comment by ccbogel** [comment]
I might have to move this to a separate issue as the original has been resolved.
I have no problems opening files with or without text transcriptions. Something else is occurring. I think I suspect that the id that links the av source to the text transcription is not there, for some reason.
In Reports , Sql Statements try this:
SELECT id, name, av_text_id FROM  source
![pic1](https://user-images.githubusercontent.com/5834638/139738870-6b35fc30-73a9-4542-a9e2-0c0fde337ce0.png)

The A/V source should have an id entry that corresponds with the text file entry, as shown in the picture.
I am guessing in your case it does not. Please let me know. So I am not sure of the original cause of this, You can delete the text file entry and that is one cause. So if this occurred, I will have to add extra checks to warn of this, and extra workarounds to resolve this too.

Regards online services, most are sign up (eg Azure, Bing, Houndify, IBM, Wit.ai), Google is free but limited. The user can to decide whether to use these services and T&Cs .

**Comment by glocalglocal** [comment]
You are right. The two mp3 files that existed before yesterday and 
generated the errors have no av_text_id value. While testing yesterday I 
linked a new mp3 file and a blank txt was automatically created. This 
works as expected and has an av_text_id value pointing to the correct txt.

A while ago, when I finished transcribing the mp3 files that now 
generated the errors I processed the txt files further outside qc and 
linked them to qc as documents (.md in this case). This was necessary 
because I wanted to add markdown formatting. That may have broken the 
link between mp3 and text files. However, when I open again the mp3 file 
I think it should open as a new mp3 file ready for transcribing. In any 
case, it shouldn't generate an error.

**Comment by ccbogel** [comment]
OK, solved for now.
If there is a missing link it tries to recreate it by looking for matching a database source.name entry e.g. my.mp3.txt. or my.mp3.transcribed. If not found, it will create a new blank database text entry in the source table.

The text and details stored in the database are what is used by QC. Despite changing the original files or linking etc.
